JamieJQuinn/IR-GTS
# Updated to use Docker

Hey @JamieJQuinn,

First, can i ask you to make a new branch called Docker and I'll update this PR to merge into that? I still have some stuff that needs to be fixed/updated but wanted to share what I had first.

I really like this project but didn't like that i needed to run it with admin so I updated a bunch of stuff to use Docker. I tested and got the TRU Arceus on my HG game.

While moving stuff around I added a few routes

These are still a work in progress:
1. /pokemondpds/worldexchange/list_options - lists all the pkm files in the /pkms/ folder. each file will have a select button next to it that makes a post request to /pokemondpds/worldexchange/select with the file name
2. /pokemondpds/worldexchange/select - accepts POST requests. when this endpoint gets a file name it starts hosting that file when you connect to the GTS
3. /pokemondpds/worldexchange/selected - returns a string with the name of the currently selected file


Currently, these changes will automatically give you the first pkm file found in the /pkms/ directory.

To test my changes you just need to
1. update the docker-compose.yml file to have your computer's IP
2. ensure ports are open
3. add any pkm files to pkms/
4. run "docker compose up"
